cachito/web/content_manifest.py

```diff
@@ -71,6 +77,21 @@ def set_go_package_sources(self):
                     flask.current_app.logger.warning("Could not find a Go module for %s", purl)
             pkg_data.pop("name")
 
+    def process_npm_package(self, package, dependency):
+        """
+        Process npm package.
+
+        :param Package package: the npm package to process
+        :param Dependency dependency: the npm package dependency to process
+        """
+        if dependency.type == "npm":
+            purl = package.to_purl()
+            self._npm_data.setdefault(purl, {"purl": purl, "dependencies": [], "sources": []})
+            icm_dependency = {"purl": dependency.to_purl()}
+            self._npm_data[purl]["sources"].append(icm_dependency)
+            if not dependency.dev:
```

Comment:
Is there a test for this?

Reply:
Yes, the test for this method has checks for `sources` and `dependencies`